csrc/scheduler/normalization_utils.cpp

```diff
@@ -727,26 +727,12 @@ int64_t getMaxRegOrSharedMemorySizeForPersistentBuffer(
   if (!can_use_smem_persistent) {
     return available_persistent_buffer_size;
   }
-  // Check available shared memory
   const auto dev_prop = at::cuda::getCurrentDeviceProperties();
-  const int64_t max_shared_memory_size =
-      (int64_t)dev_prop->sharedMemPerBlockOptin;
-  // Some shared memories are reserved for kernel launch overhead and
-  // reduction_broadcast_workspace. Estimation is conservative, but should
-  // be good enough. The actual threads per block is set in the heuristics
-  // and it may be smaller than maxThreadsPerBlock.
-  // TODO: More accurate estimation of available shared memory size.
-  const int64_t kernel_overhead = (int64_t)dev_prop->reservedSharedMemPerBlock;
-  int64_t max_buffer_dtype_size = 1;
-  for (auto tv : persistent_buffers) {
-    max_buffer_dtype_size = std::max(
-        max_buffer_dtype_size,
-        dataTypeSize(tv->getDataType().value(), runtime_info.getIndexType()));
-  }
-  const int64_t reduction_broadcast_workspace =
-      (int64_t)(dev_prop->maxThreadsPerBlock) * max_buffer_dtype_size;
-  const int64_t available_shared_memory_size =
-      max_shared_memory_size - kernel_overhead - reduction_broadcast_workspace;
```

Comment:
This part calculates avilable shared memory, however it didn't consider Welford ops, I changed to directly use `getSharedMemoryOverheadPerBlock`

Reply:
Where exactly did this part miss the Welford effect?

Reply:
Welford factor is not considered in this code when calculating `reduction_broadcast_workspace`

Reply:
So, IIUC, this PR actually solves two problems of calculating the shared memory size, the non-divisible effect and the broadcast workspace in the case of Welford. Correct?

Reply:
Yes, initially I only corrected the bug of calculating `broadcast workspace`, then CI detected a benchmark error, further check found we also need `non-divisible effect`

Reply:
PR description is revised to refelct these two fixes.